src/services/admin/AuditLogsService.ts

```diff
@@ -0,0 +1,500 @@
+            logger.error(
+              `Error occurred when getting the list of pull requests for the site ${
+                sessionData.siteName
+              } from GitHub: ${JSON.stringify(error)}`
+            )
+
+            return new AuditLogsError(
+              `Error occurred when getting the list of pull requests for the site ${sessionData.siteName}`
+            )
+          }
+        )
+          .andThen((pulls) => okAsync(pulls.filter((pull) => pull.merged_at)))
+          .andThen((pulls) =>
+            this.sitesService
+              .getBySiteName(sessionData.siteName)
+              .andThen((site) =>
+                ResultAsync.combine(
+                  pulls.map((pull) =>
+                    ResultAsync.fromPromise(
+                      this.reviewRequestService.getReviewRequest(
+                        site,
+                        pull.number
+                      ),
+                      (error) => {
+                        logger.error(
+                          `Error occurred while retrieving review request data from the database for pull request ${pull.number} of site ${sessionData.siteName}: ${error}`
+                        )
+                        return new DatabaseError(
+                          "Error occurred while retrieving review request data from the database"
+                        )
+                      }
+                    ).map((reviewRequest) => ({
+                      timestamp: new Date(pull.merged_at || ""),
+                      activity: AuditableActivity.PublishedChanges,
+                      actor:
+                        "requestor" in reviewRequest &&
+                        reviewRequest.requestor.email
+                          ? reviewRequest.requestor.email
+                          : pull.user?.login || "Unknown user",
+                      page: "",
+                      remarks: `GitHub Pull Request ID #${pull.number}`,
+                    }))
+                  )
+                )
+              )
+          )
+          .andThen((publishedChanges) =>
+            okAsync(
+              publishedChanges
+                .concat(auditLogs)
+                .sort((a, b) => a.timestamp.getTime() - b.timestamp.getTime())
+            )
+          )
+      )
+  }
+
+  getAuditLogsViaFormsg(
+    email: string,
+    repoNames: string[],
+    sinceDate: string,
+    untilDate: string,
+    formSubmissionId: string
+  ): ResultAsync<void, AuditLogsError | ForbiddenError> {
+    const emailSubject = `[Isomer] Site Audit Logs (submission ID: ${formSubmissionId})`
+    logger.info(
+      `Received request to get audit logs, submission ID: ${formSubmissionId}`
+    )
+
+    // Step 1: Check if the user exists
+    return ResultAsync.fromPromise(this.usersService.findByEmail(email), () => {
+      logger.error(
+        `Error occurred while retrieving user data from the database for email ${email}`
+      )
+
+      return new DatabaseError(
```

Comment:
are we emailing this error to anyone ah? seems like silent failure here :(

Reply:
This is handled in the final `orElse` flow, which is then sent to the requestor.